No need to run-depend on setuptools_rust and rust..

chunk 2: https://github.com/matrix-org/synapse/commit/5a7742a83308743dfd5e9a1fab751e345757aab7
to remove in 1.84
Index: pyproject.toml
--- pyproject.toml.orig
+++ pyproject.toml
@@ -217,7 +217,7 @@ pydantic = ">=1.7.4"
 # This isn't really a dev-dependency, as `poetry install --no-dev` will fail,
 # but the alternative is to add it to the main list of deps where it isn't
 # needed.
-setuptools_rust = ">=1.3"
+# setuptools_rust = ">=1.3"
 
 
 # Optional Dependencies
@@ -368,7 +368,7 @@ furo = ">=2022.12.7,<2024.0.0"
 # system changes.
 # We are happy to raise these upper bounds upon request,
 # provided we check that it's safe to do so (i.e. that CI passes).
-requires = ["poetry-core>=1.0.0,<=1.5.0", "setuptools_rust>=1.3,<=1.5.2"]
+requires = ["poetry-core>=1.0.0,<=1.5.0", "setuptools_rust>=1.3,<=1.6.0"]
 build-backend = "poetry.core.masonry.api"
 
 
